### PR TITLE
Submodules

### PR DESCRIPTION
**Pull request by rwngwn**
reopening #31 against master

### CODE REVIEW

**Comment by vrutkovs** [comment]
+1 on the code, tests need some adjustment though - we should add existing common-steps/common-steps before we do `update`

**Comment by rwngwn** [comment]
@vrutkovs fixed tests and added common steps back.

File: ctf_cli/application.py [hunk 1]
```diff
@@ -100,22 +101,13 @@ def init(self):
...
-        else:
-            logger.info("Adding tests/features/common-features as a submodule")
-            check_call('git submodule add https://github.com/Containers-Testing-Framework/common-features.git tests/features/common-features', shell=True)
-
         common_steps_dir = os.path.join(steps_dir, "common_steps")
         if os.path.exists(common_steps_dir):
             logger.info("Directory tests/steps/common_steps already exists")
         else:
             logger.info("Adding tests/steps/common_steps as a submodule")
             check_call('git submodule add https://github.com/Containers-Testing-Framework/common-steps.git tests/steps/common_steps', shell=True)
```
Comment (vrutkovs) [inline ctf_cli/application.py]: We shouldn't forcibly add `common-steps` here

File: circle.yml [hunk 2]
```diff
@@ -15,6 +15,7 @@ test:
   override:
     - mkdir test_dir
     - cd test_dir && COVERAGE_FILE=../.coverage.init coverage run ../ctf-cli.py init
+    - cd test_dir && COVAREGE_FILE=../.coverage.remote_add coverage run ../ctf-cli.py remote add features https://github.com/Containers-Testing-Framework/common-features.git
```
Comment (vrutkovs) [inline circle.yml:19]: We should add `common-steps` here instead

**Comment by rwngwn** [comment]
hmm I cant do that easily because python doesn't like - character in imports it seems. so if I add common-steps this way. it dies.
@vrutkovs can you change repo name to common_steps instead of common-steps ? :)

**Comment by vrutkovs** [comment]
`_write_environment_py` from `behave.py` should be updated - e.g. convert dashes into underscores.

Alternatively we can add a submodule with a new name (stripping non-alphanumeric chars)

**Comment by rwngwn** [comment]
@vrutkovs Ok.. will update it in _write_environment_py. you ar right its better solution

**Pull request merged**